src/app/system/configure-data-fields/data-fields/data-fields.component.ts

```diff
@@ -0,0 +1,233 @@
+        subEntity: this.subEntity,
+        entity: this.type,
+      });
+      this.rows.push(row);
+      this.updateView();
+      this.fieldName = null;
+      this.columnType = null;
+      this.addField.control.clearValidators();
+    } else {
+      this.toggleMatTable = false;
+    }
+  }
+
+  updateView() {
+    this.addDataSource.next(this.rows.controls);
+  }
+
+  removeNode(index: number, flag = false) {
+    const control = this.form.get("tableArray") as FormArray;
+    control.removeAt(index);
+    this.updateView();
+  }
+
+  changeType(index: number, value: any) {
+    const controls = this.form.get("tableArray") as FormArray;
+    if (controls) {
+      let control: any = controls.controls[index];
+      control.value.type = value;
+      if (value !== "String") {
+        control.controls["length"].disable();
+      } else {
+        control.controls["length"].enable();
+      }
+
+      if (value !== "Dropdown") {
+        control.controls["code"].disable();
+      } else {
+        control.controls["code"].enable();
+      }
+    }
+    this.updateView();
+  }
+
+  saveControls() {
+    let value = this.form.get("tableArray").value;
+
+    let appTableName=this.type.toLowerCase();
+    if(this.type.toLowerCase()==='loantype'){
+      appTableName='loan';
+    }
+    else if(this.type.toLowerCase()==='ou'){
+      appTableName='office';
+    }
+
+    let model = {
+      apptableName: `m_${appTableName}`,
+      datatableName: this.datatableName,
+      officeCountryId: this.countryId,
+      addColumns:value
+    };
+
+    this.systemService.saveFieldConfiguration(model).subscribe(
+      (res: any) => {
+        this.rows.clear();
+        this.getFieldConfigurationByType();
+        this.toggleMatTable = false;
+      },
+      (err) => {
+        let errorMessage=err?.error?.error;
+        if(err?.error?.errors){
+          errorMessage=err?.error?.errors[0]?.developerMessage;
+        }
+        this.alertService.alert({
+          type: "Error while saving data",
+          message: errorMessage,
+        });
+      }
+    );
+  }
+
+  updateControl(value: any) {
+    let model = {
+      apptableName: `m_${this.type.toLowerCase()}`,
+      datatableName: this.datatableName,
+      field: value.field,
+      isEnabled: value.isEnabled,
+      isMandatory: value.isMandatory,
+      isUnique: value.isUnique,
+    };
+    this.systemService
+      .updateFieldConfiguration(model, value.fieldConfigurationId, this.countryId)
+      .subscribe((res: any) => {
+        this.getFieldConfigurationByType();
+        this.toggleMatTable = false;
+      });
+  }
+
+  deleteRecord(field: any, fieldConfigurationId: any) {
+    let model = {
+      apptableName: `m_${this.type.toLowerCase()}`,
+      datatableName: this.datatableName,
+    };
+    this.systemService.deleteConfiguration(fieldConfigurationId, model).subscribe((res: any) => {
+      this.getFieldConfigurationByType();
+      this.toggleMatTable = false;
+    });
+  }
+
```

Comment:
Getting this error when I try to delete a custom field, and thinking that this error has something to do with the payload. consider the function I shared to generate the `apptableName` to use 
![Screenshot 2023-05-15 at 3 20 09 PM](https://github.com/one-acre-fund/fineract-frontend-v2/assets/40672143/d491784a-ba34-4333-81a4-456f9d1d1169)

Reply:
Consider clearing any error message when this is clicked... the red error message in the screenshot is something to check on even after when you hit the add button for creating a new field
<img width="1296" alt="Screenshot 2023-05-15 at 10 08 39 AM" src="https://github.com/one-acre-fund/fineract-frontend-v2/assets/40672143/85c92539-13c7-4ac9-958e-9ff409e187fc">

Reply:
Was this fixed @william000000